tests/regression/src/lib.rs

```diff
@@ -55,18 +103,29 @@ mod tests {
             cg::cachegrind::start_instrumentation();
         }
     }
+
     /// Environment variable to determine whether to run under valgrind or solely test functionality.
     fn is_running_under_valgrind() -> bool {
         env::var("ENABLE_VALGRIND").is_ok()
     }
 
+    /// Function to determine if diff mode is enabled, runs diff assertion if it is.
+    fn is_diff_mode() -> bool {
+        env::var("DIFF_MODE").is_ok()
+    }
+
     fn valgrind_test<F>(test_name: &str, test_body: F) -> Result<(), s2n_tls::error::Error>
     where
         F: FnOnce(&InstrumentationControl) -> Result<(), s2n_tls::error::Error>,
     {
         if !is_running_under_valgrind() {
-            let ctrl = InstrumentationControl;
-            test_body(&ctrl)
+            if is_diff_mode() {
```

Comment:
I wonder if using `enums` might might make this a little neater? You could have a single enum for `RegressionTestMode` to match on. That would also reduce the number of `is_*_mode()` functions needed.